deps/github.com/arangodb/arangosync/client/api.go

```diff
@@ -0,0 +1,333 @@
+type MasterInfo struct {
+	// Unique identifier of the master
+	ID string `json:"id"`
+	// Internal endpoint of the master
+	Endpoint string `json:"endpoint"`
+	// Is this master the current leader
+	Leader bool `json:"leader"`
+}
+
+type RoleInfo struct {
+	Role Role `json:"role"`
+}
+
+type Role string
+
+const (
+	RoleMaster Role = "master"
+	RoleWorker Role = "worker"
+)
+
+func (r Role) IsMaster() bool { return r == RoleMaster }
+func (r Role) IsWorker() bool { return r == RoleWorker }
+
+type ChannelPrefixInfo struct {
+	Prefix string `json:"prefix"`
+}
+
+// SyncInfo holds the JSON info returned from `GET /_api/sync`
+type SyncInfo struct {
+	Source         Endpoint           `json:"source"`                   // Endpoint of sync master on remote cluster
+	Status         SyncStatus         `json:"status"`                   // Overall status of (incoming) synchronization
+	Shards         []ShardSyncInfo    `json:"shards,omitempty"`         // Status of incoming synchronization per shard
+	Outgoing       []OutgoingSyncInfo `json:"outgoing,omitempty"`       // Status of outgoing synchronization
+	MessageTimeout time.Duration      `json:"messageTimeout,omitempty"` // Maximum time between messages in a task channel
+}
+
+// OutgoingSyncInfo holds JSON info returned as part of `GET /_api/sync`
+// regarding a specific target for outgoing synchronization data.
+type OutgoingSyncInfo struct {
+	ID       string          `json:"id"`               // ID of sync master to which data is being send
+	Endpoint Endpoint        `json:"endpoint"`         // Endpoint of sync masters to which data is being send
+	Status   SyncStatus      `json:"status"`           // Overall status for this outgoing target
+	Shards   []ShardSyncInfo `json:"shards,omitempty"` // Status of outgoing synchronization per shard for this target
+}
+
+// ShardSyncInfo holds JSON info returned as part of `GET /_api/sync`
+// regarding a specific shard.
+type ShardSyncInfo struct {
+	Database              string        `json:"database"`                 // Database containing the collection - shard
+	Collection            string        `json:"collection"`               // Collection containing the shard
+	ShardIndex            int           `json:"shardIndex"`               // Index of the shard (0..)
+	Status                SyncStatus    `json:"status"`                   // Status of this shard
+	StatusMessage         string        `json:"status_message,omitempty"` // Human readable message about the status of this shard
+	Delay                 time.Duration `json:"delay,omitempty"`          // Delay between other datacenter and us.
+	LastMessage           time.Time     `json:"last_message"`             // Time of last message received by the task handling this shard
+	LastDataChange        time.Time     `json:"last_data_change"`         // Time of last message that resulted in a data change, received by the task handling this shard
```

Comment:
... and this is C style underscore.  Is there a reason for different styles?  Jan gets upset if I use C style.